packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -595,7 +595,7 @@ class FocusNode with DiagnosticableTreeMixin, ChangeNotifier {
   ///
   /// Use [nearestScope] to start at this node instead of above it.
   FocusScopeNode get enclosingScope {
-    return ancestors.firstWhere((FocusNode node) => node is FocusScopeNode, orElse: () => null);
+    return ancestors.whereType<FocusScopeNode>().firstWhere((_) => true, orElse: () => null);
```

Comment:
The `firstWhere((_) => true` part is strange.

Reply:
Yes, but `.first` throws if the Iterable is empty. `firstWhere` is used only to have a default to null.

Here are the alternatives:

```dart

// using a explicit cast
FocusScopeNode get enclosingScope {
  return ancestors.firstWhere((FocusNode node) => node is FocusScopeNode, orElse: () => null) as FocusScopeNode;
}

// using firstWhere and orElse
FocusScopeNode get enclosingScope {
  return ancestors.whereType<FocusScopeNode>().firstWhere((_) => true, orElse: () => null);
}

// catching error
FocusScopeNode get enclosingScope {
  try {
    return ancestors.whereType<FocusScopeNode>().first;
  } on StateError {
    return null;
  }
}

// with spread
FocusScopeNode get enclosingScope {
  return <FocusScopeNode>[...ancestors.whereType<FocusScopeNode>(), null].first;
}

// with static extension methods
extension on Iterable<T> {
  T get firstOrNull  {
    try {
      return this.first;
    } on StateError {
      return null;
    }
  }
}
FocusScopeNode get enclosingScope {
  return ancestors.whereType<FocusScopeNode>().firstOrNull;
}
```

What would you prefer?

Reply:
They are all not great, I would prefer the "using a explicit cast" option, though.

Reply:
Changed with "using a explicit cast"